src/health_checker.cc

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2018 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+#include "health_checker.h"
+
+#include <boost/filesystem.hpp>
+#include <boost/algorithm/string/join.hpp>
+#include <vector>
+#include <fstream>
+
+namespace google {
+
+HealthChecker::HealthChecker(const MetadataAgentConfiguration& config)
+    : config_(config) {
+  boost::filesystem::create_directories(
+      boost::filesystem::path(config_.HealthCheckFile()).parent_path());
+  std::remove(config_.HealthCheckFile().c_str());
```

Comment:
I assume this is just defensive programming in case we are running in space with a non-ephemeral filesystem?

Reply:
Yes. We don't want a brand new agent process to start out unhealthy.

Reply:
Correct.  We want to make sure the "unhealthy" flag is not set at startup.

Reply:
SGTM